### PR TITLE
GL-1483.  Pull Fingerprints from Mercury

### PR DESCRIPTION
**Pull request by gbggrant**
Modified DownloadGenotypes and UploadFingerprintJsonToMercury to
write to the Mercury web service from the cloud.
Pull credentials from vault
Added Several Scientific tests to exercise the fingerprint download/upload functionality.
Modified the test runner to only pull test files that have the extension '.json'

### CODE REVIEW

**Review by jessicaway — APPROVED** [review]
A few minor comments, but looks good overall

File: tests/broad/scala_test/src/main/scala/org/broadinstitute/dsp/pipelines/tester/ValidationWdlTester.scala [hunk 1]
```diff
@@ -200,19 +200,23 @@ abstract class ValidationWdlTester(testerConfig: BaseConfig)(
...
-      val truthPath = truthPrefix.resolve(s"$inputsName/")
-
-      WorkflowRunParameters(
-        id = s"${envString}_$inputsName",
-        workflowInputs = getInputContents(fileName),
-        resultsCloudPath = resultsPath,
-        truthCloudPath = truthPath
-      )
-    }
+    workflowInputRoot.list
```
[the remark this answers is not in the document]
Comment (gbggrant) [reply tests/broad/scala_test/src/main/scala/org/broadinstitute/dsp/pipelines/tester/ValidationWdlTester.scala:203]: No, I added the .filter on line 204 to have it only use .json files as test inputs.  Not sure that's the best scala, but it's a real change.

File: tasks/broad/IlluminaGenotypingArrayTasks.wdl [hunk 2]
```diff
@@ -122,11 +124,14 @@ task GtcToVcf {
...
     --ANALYSIS_VERSION_NUMBER ~{analysis_version_number} \
     --EXPECTED_GENDER "~{reported_gender}" \
-    ~{"--FINGERPRINT_GENOTYPES_VCF_FILE " + fingerprint_genotypes_vcf_file} \
+    ~{fp_vcf_arg} \
     --REFERENCE_SEQUENCE ~{ref_fasta} \
     --MAX_RECORDS_IN_RAM 100000 \
     --CREATE_INDEX true \
     --PIPELINE_VERSION ~{pipeline_version}
+
+    #    ~{"--FINGERPRINT_GENOTYPES_VCF_FILE " + fingerprint_genotypes_vcf_file} \
```
[the remark this answers is not in the document]
Comment (gbggrant) [reply tasks/broad/IlluminaGenotypingArrayTasks.wdl]: Sorry - cleaned that up.

**Review by khajoue2 — APPROVED** [review]
lgtm

**Pull request merged**